src/library_browser.js

```diff
@@ -1291,37 +1291,37 @@ var LibraryBrowser = {
     return info.awaited;
   },
 
-  emscripten_get_preloaded_image_data__deps: ['$PATH_FS', 'malloc'],
+  emscripten_get_preloaded_image_data__deps: ['$getPreloadedImageData', '$UTF8ToString'],
   emscripten_get_preloaded_image_data__proxy: 'sync',
-  emscripten_get_preloaded_image_data: (path, w, h) => {
-    if ((path | 0) === path) path = UTF8ToString(path);
```

Comment:
This could also have been fixed with `typeof path !== 'string'`, but I think your approach is cleaner..? lgtm either way.

Reply:
Yeah, I think this approach is better since it avoids the check completely in the default path.